### PR TITLE
Implement first basic version of batch spec execution UI

### PR DESCRIPTION
**Pull request by eseliger**
First version of some sort of UI, so we can debug easier. This is neither complete nor got any fancy designs, but it does for now. Will follow-up in separate PRs with more information and some higher quality UI.

For now, here's some exciting progress:

image

### CODE REVIEW

**Review by LawnGnome — APPROVED** [review]
LGTM.

File: client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                className: 'bg-success',
+            },
+
+            execution.state === BatchSpecExecutionState.COMPLETED
+                ? { icon: <CheckIcon />, text: 'Finished', date: execution.finishedAt, className: 'bg-success' }
+                : { icon: <ErrorIcon />, text: 'Failed', date: execution.finishedAt, className: 'bg-danger' },
+        ],
+        [execution]
+    )
+    return <Timeline stages={stages.filter(isDefined)} now={now} className={className} />
```
Comment (courier-new) [inline client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:133]: Ugh I LOVE that this is already a thing!! Hadn't realized this before, thank you for bringing this to my attention! 🤩

File: enterprise/internal/batches/resolvers/resolver_test.go [hunk 2]
```diff
@@ -1505,7 +1505,8 @@ func TestResolver_CreateBatchSpecExecution(t *testing.T) {
 
 	ctx := context.Background()
 	db := dbtest.NewDB(t, "")
-	cstore := store.New(db, nil)
+	now := time.Now().UTC().Truncate(time.Millisecond)
+	cstore := store.NewWithClock(db, nil, func() time.Time { return now })
```
Comment (courier-new) [inline enterprise/internal/batches/resolvers/resolver_test.go:1509]: Neat!

**Review by courier-new — APPROVED** [review]
Also looks good to me 😄

File: enterprise/cmd/executor-queue/internal/queues/batches/transform.go [hunk 3]
```diff
@@ -110,7 +110,7 @@ const (
 )
 
 func createAccessToken(ctx context.Context, db dbutil.DB, userID int32) (string, error) {
-	_, token, err := database.AccessTokens(db).Create(ctx, userID, []string{accessTokenNote}, accessTokenNote, userID)
+	_, token, err := database.AccessTokens(db).Create(ctx, userID, []string{accessTokenScope}, accessTokenNote, userID)
```
Comment (mrnugget) [inline enterprise/cmd/executor-queue/internal/queues/batches/transform.go:113]: Thank you! :bow:

**Review by mrnugget — APPROVED** [review]

**Pull request merged**